### DISCUSSION TITLE
Offline javascript resources

### DISCUSSION THREAD

**Discussion by murrant** [php-flasher/php-flasher#11] [Q&A]
What is the best way to use the javascript libraries offline.

My current guess is download the js, and include it in the page.  Set the scripts arrays in the config files to empty.
(We don't know the url the application will be at)

**Comment by yoeunes** [comment]
I forget to document this part 👀 sorry.

your guess is correct:

>  My current guess is download the js, and include it in the page. Set the scripts arrays in the config files to empty.

Or you can install them with npm, check this here : https://www.npmjs.com/search?q=php-flasher .

Just be aware that you js files will be included on every page, if you really care about the page size:

> 1. download the js files you want
> 2. put them in the public folder
> 3. replace the cdn url

and this way if there is not notification in the storage (session), the files will not be included in the page

**Reply by murrant** [reply]
I just added them to our page layout and deleted the "scripts" key on all templates.

I can't set in config files because it might be wrong for users loading it, using the asset() function makes sure the url is correct.
